### ISSUE TITLE
Changing Month (Jan->Feb) on Jan 29 Crashes both 5.2.4 and 5.2.6 (W10)

### ISSUE THREAD

**Issue by soggygeek** [seandepagnier/climatology_pi#82]
I have 2 computers, both running W10.  Both were running the latest version of Climatology from the plugin manager.  (1.4.39)  One is running 5.2.4 and one 5.2.6 OpenCPN.  Both of them crash today on changing the month from January to February.  (perhaps because the date is currently set to 29?)
I downgraded one computer from Climatology 1.4.39 to 1.4.26, as this had fixed a problem I had with Weather Routing about 10 days ago.  It did not fix this problem.
Then I changed the date from Jan 29 to Jan 26 and THEN changed the month from Jan to Feb, and it did not crash.
Seems a fairly significant bug but maybe only with January->February with dates > 28.
Thanks.

**Comment by rgleason** [comment]
Where exactly are you changing this date?  The days are not available in Climatology, only the months.

**Comment by LeaseOnLife** [comment]
This is a duplicate, I reported  that last year and it got fixed/closed:
https://github.com/seandepagnier/climatology_pi/issues/76

I compile from git, not sure whether that has made its way into the OpenCPN catalog

**Comment by rgleason** [comment]
No, this has not been done, note that Bdbcat made changes before that here for MacOS
rgleason@6c4980f

**Comment by rgleason** [comment]
LeaseOnLife wrote
```
This is a duplicate, I reported that last year and it got fixed/closed:
https://github.com/seandepagnier/climatology_pi/issues/76

I compile from git, not sure whether that has made its way into the OpenCPN catalog
```
Thanks I found that change by Sean and have tried it but it fails on windows. OpenCPN freezes.

I can also confirm that this date problem exists still in version 1.4.39  although I cannot see the "day" because it is obsured by the up/down arrows.  But I have been able to make the plugin fail switching from Jan to Feb after hitting the up button to change days.

**Comment by rgleason** [comment]
I have updated my repos for  Testplugin (TP) Frontend 1.0.193.0 so that we get updated OS builds for RPI and Flatpak.
This is working, except for the two problems noted here in issues.
  A. High res displays don't show the "day" 
  B. Changing the month from January with the day on "31"  to February causes O crash.

We need a solution for these problems.

**Comment by rgleason** [comment]
I think I have isolated the problem to when  only "Current" is checked and it is set to January and you try to scroll up or down through January.  Then it freezes.  Then try this on any month and it fails.  Current is very buggy!  Why?

-  When  only "Wind" is checked it seems to work.  Set it on any month and hit the up or down button, then switch months.  Everything is ok, no freeze
- When only "Cyclones" is checked it seems to work. Set it on any month and hit the up or down button, then switch months. Everything is ok, no freeze
-

**Comment by rgleason** [comment]
Sean, I have updated the plugin to version 1.4.40 using frontend TP 1.0.193 and Jon has fixed it to build flatpak.  

I believe I have fixed one of the date problems, see your [commit](https://github.com/seandepagnier/climatology_pi/commit/15d410124286d3c9718ec865662b5fddf141da9e) (that I also referred to),  so that it works with Windows,  Your change did not seem to work in windows. I have left comments in [climatologyDialog.cpp at lines 235-246 ](https://github.com/rgleason/climatology_pi/blob/9ed7bd5e176426608b4af6a143246c3c68c9887d/src/ClimatologyDialog.cpp) for your review.

However there still exists another date problem which is the subject of this Issue. I can confirm that Opencpn freezes when

1. Climatology is showing "Currents" checked.
2. Date is changed to January 29 (or maybe 30) (I think, because I can't see the date on my high res screen).
3. Then the date is changed to February.
4. The plugin freezes and and then Opencpn fails.
5. Please see my tests confirming and clarifying this Issue #84

Can you fix this problem?

Also there is another problem on high resolution displays which is a problem.
 High Resolution Screens do not show the "Days" #83 
I have tried using wxformbuilder on a separate test repos and could not make it work right.
Perhaps you can fix this so that pulling the menu window wider will expose the "Days"?
I could not get it to work.

**Comment by rgleason** [comment]
I am in the process of incrementing to version 1.4.41 and releasing after some tries to fix this.

I have once again checked this problem.  

1. Note that the problem does **not exist** when doing the same thing, but just checking Wind or Cyclones.
2. When I check "Currents" 
3. Month set to January.
4. Using the Up button  scroll up to "29" or maybe "30" (I can't see the date on my hires screen).
5. When it hits that date or the Month is then changed to "February" the plugin freezes and OpenCPN FAILS.

Also need to be able to increase the diameter of the wind rose so we can see the average wind in side that!
https://github.com/rgleason/climatology_pi

**Comment by rgleason** [comment]
Now 2026,  I think most of this is fixed and if not is in rgleason Issues.
